terraform/azure/variables.tf

```diff
@@ -92,27 +78,47 @@ variable "ssh_pub_key" {
   EOT
 }
 
-variable "notebook_nodes" {
+variable "core_node_pool" {
+  type = object({
+    name : optional(string, ""),
```

Comment:
Since `""` is not a valid name, I think this should be marked as required rather than optional (same as `vm_size`)